src/main/java/org/opensearch/neuralsearch/processor/InferenceProcessor.java

```diff
@@ -165,6 +180,31 @@ public void batchExecute(List<IngestDocumentWrapper> ingestDocumentWrappers, Con
         });
     }
 
+    private Tuple<List<String>, Map<Integer, Integer>> sortByLengthAndReturnOriginalOrder(List<String> inferenceList) {
+        List<Tuple<Integer, String>> docsWithIndex = new ArrayList<>();
+        for (int i = 0; i < inferenceList.size(); ++i) {
+            docsWithIndex.add(Tuple.tuple(i, inferenceList.get(i)));
+        }
+        docsWithIndex.sort(Comparator.comparingInt(t -> t.v2().length()));
+        List<String> sortedInferenceList = docsWithIndex.stream().map(Tuple::v2).collect(Collectors.toList());
+        Map<Integer, Integer> originalOrderMap = new HashMap<>();
+        for (int i = 0; i < docsWithIndex.size(); ++i) {
+            originalOrderMap.put(i, docsWithIndex.get(i).v1());
+        }
+        return Tuple.tuple(sortedInferenceList, originalOrderMap);
+    }
+
+    private List<?> restoreToOriginalOrder(List<?> results, Map<Integer, Integer> originalOrder) {
+        List<Object> sortedResults = new ArrayList<>();
+        sortedResults.addAll(results);
+        for (int i = 0; i < results.size(); ++i) {
+            if (!originalOrder.containsKey(i)) continue;
```

Comment:
This condition won't be triggered, right?

Reply:
Actually, ml-commons could return more than what we send (which I will fix later) so it could be triggered. For example, we have docs [a, b, c, d, e], the return could be [ra, rb, rc, rd, re, rc, rd, re, re], the good thing is it wont' cause incorrect response to user.

Reply:
I also noticed this problem, and fixed it in https://github.com/opensearch-project/ml-commons/pull/2462. After that I think this won't be triggered